source/how-tos/boot-micropython.rst

```diff
@@ -0,0 +1,110 @@
+.. _boot-micropython:
+
+Boot MicroPython
+-----------------
+
+|MicroPython| is a lean and efficient implementation of the Python 3 programming language
+that includes a small subset of the Python standard library. I can be used to run python
+based applications in SBL environment.
+
+Below are the steps to enable MicroPython for SBL QEMU platform. Similar steps can be
+followed to enable it on other platforms.
+
+.. |MicroPython| raw:: html
+
+   <a href="https://docs.zephyrproject.org/latest/getting_started/getting_started.html" target="_blank">MicroPython</a>
+
+How to Build MicroPython Payload Module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Clone latest SBL tree and prepare the build environment following :ref:`build-steps`.
+   Here, assume SBL is cloned into directionary named 'SlimBoot'. ::
+
+     git clone https://github.com/slimbootloader/slimbootloader.git SlimBoot
+
+#. Clone MicroPython payloads repo into SBL tree root directory named 'PayloadModPkg'. ::
+
+     cd SlimBoot
+     git clone https://github.com/slimbootloader/payloads.git PayloadModPkg
+     cd PayloadModPkg
+     git submodule update --init
+     cd ..
+     python PayloadModPkg\MicroPython\Tools\prep.py
+
+#. Build MicroPython module binary.
+   Currently, only Windows VS2017 Community build environment is tested for MicroPython build. ::
+
+     python BuildLoader.py build_dsc -p PayloadModPkg\PayloadModPkg.dsc -d MICRO_PYTHON
+
+#. Copy generated MicroPython payload module binary to payload binary directory.
+   The MicroPython.efi file path might change depending on the selected build toolchain. ::
+
+     mkdir PayloadPkg\PayloadBins
+     copy Build\PayloadModPkg\DEBUG_VS2017\IA32\MicroPython.efi PayloadPkg\PayloadBins /y
+
+
+How to Integrate MicroPython into Slim Bootloader
```

Comment:
Should we switch the order? The generic model first and then a specific use of enabling setup using micropython next? I am ok with this order as well.

Reply:
Yes, the reason I talk about the setup first because we have an existing full example already in SBL.  And it is easier to follow.   For other use cases, it might need additional modifications.

Reply:
Sounds good.